src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -26,17 +26,20 @@ public class DbConnectionValidator : IDbConnectionValidator
     private static readonly string _shellDescriptorTypeColumnValue = new TypeService()[typeof(ShellDescriptor)];
 
     private readonly IEnumerable<DatabaseProvider> _databaseProviders;
+    private readonly IConfiguration _yesSqlConfiguration;
     private readonly YesSqlOptions _yesSqlOptions;
     private readonly SqliteOptions _sqliteOptions;
     private readonly ShellOptions _shellOptions;
 
     public DbConnectionValidator(
         IEnumerable<DatabaseProvider> databaseProviders,
+        IConfiguration yesSqlConfiguration,
```

Comment:
Weird to need both "YesSqlOptions" and "YeslSqlConfiguration"

Reply:
`YesSqlOptions` has the option [TablePrefixSeparator ](https://github.com/OrchardCMS/OrchardCore/blob/8d9ced24fe9c815ccbc309cdf2a48db37db6a1aa/src/OrchardCore/OrchardCore.Data.YesSql.Abstractions/YesSqlOptions.cs#L21) which isn't used in `YesSql`. This is the reason both are needed. I think we should use `IConfiguration` to get the currently configured services instead of using `YesSqlOptions` since `YesSqlOptions` could have null configurations like a null value for `TableNameConvention`

Reply:
I may be wrong but `YesSqlOptions` provides options to configure the yesSql store configuration.

So why not just usin the `IStore` to retrieve what is actually used.

From memory, I did a PR to ensure that `TableNameConvention` is in sync

                // Ensure a non null 'TableNameConvention' to be always used for `YesSql.Configuration` and then in sync with it.
                services.PostConfigure<YesSqlOptions>(o => o.TableNameConvention ??= new YesSql.Configuration().TableNameConvention);

For `TablePrefixSeparator` it has a default value, and it is used indirectly by YesSql through the TablePrefix.

                    if (!String.IsNullOrWhiteSpace(shellSettings["TablePrefix"]))
                    {
                        var tablePrefix = shellSettings["TablePrefix"].Trim() + yesSqlOptions.TablePrefixSeparator;
                        storeConfiguration = storeConfiguration.SetTablePrefix(tablePrefix);
                    }

Reply:
Yes `TableNameConvention` should be retrieved from the `ISession.Store.Configuration` not from `YesSqlOptions`. But we should still use `YesSqlOptions` to determine the prefix-separator that the tenant is using to determine if the table exists or not

Reply:
I don't agree, normally we could use the table prefix configured on yesSql side, it should have already use the same configured separator on OC side (or its default value `_`).

@hishamco for your question I saw somewhere, yesSql IConfiguration is not registered in the DI.

Reply:
In summary as I can see ATM, `YesSqlOptions` should be only used to configure the store, then we can always retrieve values from the store.Configuration, unless there is no store registered yet, as we use for the tenant validator.

Reply:
Oops, and tenant validator uses `IDbConnectionValidator` so looks like here the `IStore` can't be injected, and that all need to retrived from `YesSqlOptions` that may need to be completed ;)

Reply:
@jtkech I don't think the prefix table separator is a configurable value in YesSql. AFAIK, you can't retrieve it from YesSql. I think you can get the prefix from YesSql which is not tableSeperator. TableSeperator is something configurable is OC and could be different from one tenant to another. However, in the DbConnectionValidator we assume that the connection being validated will use the currently configured table separator.

Reply:
> I don't think the prefix table separator is a configurable value in YesSql

Yes, I was just meaning that the yesSql store config TablePrefix includes it, but if here we are in the Default tenant to validate the connection of a new tenant, we have to use the tablePrefix param passed to the `ValidateAsync()` and then yes the separator from the YesSql options.

But here we will get the table separator from the YesSqlOptions related to the Default tenant, that's okay if we assume that we only configure these options for the Default tenant, and because we don't use this separator as a tenant settings in the setup form / settings editor.

> TableSeperator is something configurable is OC and could be different from one tenant to another.

Oops ;)

> However, in the DbConnectionValidator we assume that the connection being validated will use the currently configured table separator.

Yes, but you see the little issue.

Just saw that there is a new option field `UseLegacyIdentityColumn`, same remarks, e.g. if it is selectable in the new setup screens, tenant settings editor, okay we will be able to pass it as a new param to `ValidateAsync()`, and in that case not part of the YesSqlOptions (or yes just to show/use a default). Note:  For the separator, one solution would be to do the same. Otherwise here same little issue.

Hmm, maybe here the right solution is to use the dotnet `IConfiguration` to retrieve the `YesSqlOptions` from the config stack but under the specific section of the new tenant we want to create and of which we know the name.

Reply:
@jtkech for your [comment](https://github.com/OrchardCMS/OrchardCore/pull/12528#discussion_r993979250) we referring to `TablePrefixSeparator` not `TablePrefix`

Reply:
Okay, so if we can have different option values per tenant, and because here we may validate the connection of a tenant that doesn't exist yet, we need to find a way to retrieve the possible configured values for this tenant.

ShellSettings holds tenant settings and a tenant config (the injectable ShellConfiguration), for a new tenant the config values are pre-populated (here only internally) but not yet from an eventual tenant specific section.

Hmm, it may become a little more complex, if we use `IConfiguration` by specifying an eventual tenant specific section, if this section doesn't exist it will not inherit of the values of the parent section.

So looks like we need to use `IShellSettingsManager.LoadSettingsAsync(string tenant)`, look at the code to see how we stack the config sources. Note: Needs to be tried and see if it doesn't fail.

Note: Hmm, for sqlite options make sense that they are only global options, about using connexion pool or not.

Finally, for options that will be part of the setup form / tenant editor, better to manage them through the tenant settings, the ones we can retrieve by using shellSettings[""] from the injectable ShellSettings. They use the same kind of config sources as pre-configured values e.g. to pre-populate the setup form / tenant editor, and on saving updated in the tenant specific appsettings that will always win.

So in that case we would not retrieve these values from YesSqlOptions but from the provided form data (or any other binding e.g. for Web Api), as I think we already do for the pre-configurable tenant TablePrefix. And then when configuring the store we would need to use shellSettings[""] to retrieve the actual values, as for TablePrefix here also.

Sorry for the long comments, not so obvious as I need to retrieve myself how it works. Yes would be less complex if we could assume that these yesSql options are the same for all tenants, if not no luck ;)

Reply:
@jtkech @hishamco i think for now so we don’t block this PR, we should just worry about adding this new setting to OC. We can create other PR for later version on OC along with support for Schema. I suggest we only get the new version on YesSql working.
I think the DbConnectionValidator should be refactored one more time to accept ValidationRequestOptions where the consumer would provide the connection string along with other parameters like TablePrefix and TablePrefixSeperator and TableNameConvention.

Reply:
No problem to find a quick compromise to not block this PR.

Then we will have to confirm if YesSqlOptions can be per tenant or not, in both case will be quite easy to do now that we better understand/remember what we already have, this without adding additional options classes.

Reply:
@sebastienros what's the action needed here to merge this PR for `1.5` shall we inject both interfaces for now, then refactor this class later on, or shall we need to do other thing

Reply:
I will pass a `null` for the schema now, and leave everything as it was